src/plugins/expressions/common/execution/execution.ts

```diff
@@ -190,23 +197,55 @@ export class Execution<
       }
 
       const { function: fnName, arguments: fnArgs } = link;
-      const fnDef = getByAlias(this.state.get().functions, fnName);
+      const fn = getByAlias(this.state.get().functions, fnName);
 
-      if (!fnDef) {
+      if (!fn) {
         return createError({ message: `Function ${fnName} could not be found.` });
       }
 
+      let args: Record<string, ExpressionValue> = {};
+      let timeStart: number | undefined;
+
       try {
-        // Resolve arguments before passing to function
-        // resolveArgs returns an object because the arguments themselves might
-        // actually have a 'then' function which would be treated as a promise
-        const { resolvedArgs } = await this.resolveArgs(fnDef, input, fnArgs);
-        const output = await this.invokeFunction(fnDef, input, resolvedArgs);
+        // `resolveArgs` returns an object because the arguments themselves might
+        // actually have a `then` function which would be treated as a `Promise`.
+        const { resolvedArgs } = await this.resolveArgs(fn, input, fnArgs);
+        args = resolvedArgs;
+        timeStart = this.params.debug ? performance.now() : 0;
```

Comment:
should time start involve the time it took to resolve the arguments ?

Reply:
I did not include argument resolution into timing because if argument is a const it will resolve instantly, if argument is a sub-expression its execution time will also be measured but separately&mdash;it will have its own `.debug` object with all the data and timing.

But in general: (1) we can leave it as is&mdash;measure only length of function execution; or (2) we can measure two things separately and report two number, arg resolution and function duration; or (3) we can measure just one number that includes argument resolution together with function duration, but in that case I'm not sure how useful that will be for a consumer, for example, if they would want to see only how long the function executes, they would need to loop through all sub-expressions and subtract execution time of each sub-expression.

I guess the best is (2) where we report 2 number: arg resolution time and total time? WDYT?

Reply:
i think for now 1) is fine